Lib/test/test_list.py

```diff
@@ -196,6 +196,15 @@ def test_preallocation(self):
         self.assertEqual(iter_size, sys.getsizeof(list([0] * 10)))
         self.assertEqual(iter_size, sys.getsizeof(list(range(10))))
 
+    @cpython_only
+    def test_overallocation(self):
+        iterable = [1,2]
+        self.assertEqual(sys.getsizeof(iterable), sys.getsizeof(list(iterable)))
+
+        # bpo-43574: Don't overallocate for list literals
+        iterable = [1,2,3]
+        self.assertEqual(sys.getsizeof(iterable), sys.getsizeof(list(iterable)))
```

Comment:
Is there a reason you test with two different hardcoded literals?

Reply:
It's really just a minimal set of test conditions to show this specific regression; the behavior change starts with len-3 literals, as len-2 and below are special-cased to use BUILD_LIST: https://github.com/python/cpython/blob/9a50ef43e42ee32450a81ce13ed5a0729d3b84e8/Python/compile.c#L3721

I'll see if I can generalize those tests a bit more without going overboard; probably to test that initializing a new list from a known-length iterable doesn't over-allocate (ie. catching this if it regresses again for other cases), and also maybe confirming over-allocation when expected.

I'll also take care of requests 1 and 2 (CLA and NEWS item)